prog/vm/github_runner.rb

```diff
@@ -184,6 +184,16 @@ def before_run
       # ./env.sh sets some variables for runner to run properly
       ./actions-runner/env.sh
 
+      # The runner script loads environment variables from .env file but ignores
+      # the standard /etc/environment file of the operating system.
+      # https://github.com/actions/runner/issues/1703
+      cat <<EOT >> ./actions-runner/run-withenv.sh
+      #!/bin/bash
+      mapfile -t env </etc/environment
```

Comment:
til mapfile, I've often used the not-always-available `sponge` program for stuff like this (I don't write a lot of bash)

Reply:
I was not previously aware of `mapfile`. I decided to use it after it was recommended in [the issue](https://github.com/actions/runner/issues/1703). Additionally, [the GitHub k8s controller uses it](https://github.com/actions/actions-runner-controller/blob/fe8c3bb789abf7ce1bea598c47bf5a87f6adc225/runner/startup.sh#L173).